http/codegen/client_cli.go

```diff
@@ -139,14 +39,30 @@ func ClientCLIFiles(genpkg string, root *expr.RootExpr) []*codegen.File {
 		files = append(files, endpointParser(genpkg, root, svr, data))
 	}
 	for i, svc := range svcs {
-		files = append(files, payloadBuilders(genpkg, svc, data[i]))
+		files = append(files, payloadBuilders(genpkg, svc.Name(), data[i]))
 	}
 	return files
 }
 
+func buildCommandData(sd *ServiceData) *server.CommandData {
+	return server.BuildCommandData(sd.Service.Name, sd.Service.Description, sd.Service.PkgName, streamingEndpointExists(sd))
+}
+
+func buildSubcommandData(sd *ServiceData, e *EndpointData) *server.SubcommandData {
+	flags, buildFunction := buildFlags(sd, e)
+
+	sub := server.BuildSubcommandData(sd.Service.Name, e.Method, buildFunction, flags)
+	if e.MultipartRequestEncoder != nil {
```

Comment:
Here's what we can do (this is a common Go way of doing it). Let's have a struct in HTTP 
```
type commandData {
  *cli.CommandData
  Subcommands []*subcommandData
  NeedStream bool
}

type subcommandData {
  *cli.SubcommandData
  // MultipartFuncName is the function name used to render a multipart request encoder.
  MultipartFuncName string
  // MultipartFuncName is the variable name used to render a multipart request encoder.
  MultipartVarName string
}
```
So the `buildCommandData` would do
```
func buildCommandData(sd *ServiceData) *commandData {
  return &commandData{
    CommandData: server.BuildCommandData(sd.Service),
    NeedStream: streamingEndpointExists(sd),
  }
}
```
And `buildSubcommandData` would do
```
func buildSubcommandData(sd *ServiceData, e *EndpointData) *subcommandData {
   sub := &subcommandData{
     SubcommandData: cli.BuildSubcommandData(sd.Service.Name, e.Method, buildFunction, flags),
   }
   if e.MultipartRequestEncoder != nil {
      sub.MultipartVarName = e.MultipartRequestEncoder.VarName
      sub.MultipartFuncName = e.MultipartRequestEncoder.FuncName
   }
   return sub
}